### PR TITLE
chore: [IA-531] Zendesk workunit

### PR DESCRIPTION
**Pull request by debiff**
## Short description
This PR add the workunit needed to invoke the Zendesk support.
It also allow to invoke the flow instead of the instabug flow if the FF is true.

## List of changes proposed in this pull request
- Added the workunit
- Added the stub screens
- Removed the test entry point in BaseHeader component
- Routed the "?" press behaviour based on the zendesk FF

https://user-images.githubusercontent.com/11773070/144440623-1c5d7895-f1e7-480a-a046-de39ce3c8f8c.mp4

## How to test
Enable the `zendeskEnabled` FF and press the "?" button both as authenticated and not authenticated.

### CODE REVIEW

File: ts/components/ZendeskSupportComponent.tsx [hunk 1]
```diff
@@ -67,30 +73,27 @@ const ZendeskChatComponent = () => {
...
-        testID={"zendeskShowTicketsButton"}
-      >
-        <IconFont
-          name="io-chat"
-          color={IOColors.red}
-          testID={"zendeskShowTicketsIcon"}
-        />
-      </ButtonDefaultOpacity>
+      <ButtonWithImage
+        icon={""}
```
Comment (Undermaken) [inline ts/components/ZendeskSupportComponent.tsx:77]: The icon will be added later?
In any case we should use the appropriate component with the current available data
This is the same but using `ButtonDefaultOpacity`

```typescript
<View>
      <ButtonDefaultOpacity
        style={{
          alignSelf: "stretch"
        }}
        onPress={() => navigation.navigate(navigateToZendeskAskPermissions())}
        disabled={false}
        testID={"contactSupportButton"}
      >
        <Label color={"white"}>
          {I18n.t("support.helpCenter.cta.contactSupport")}
        </Label>
      </ButtonDefaultOpacity>

      <ButtonDefaultOpacity
        onPress={() => {
          showSupportTickets();
          workUnitCompleted();
        }}
        style={{
          alignSelf: "stretch"
        }}
        disabled={false}
        bordered={true}
        testID={"showTicketsButton"}
      >
        <Label>{I18n.t("support.helpCenter.cta.seeReports")}</Label>
      </ButtonDefaultOpacity>
    </View>
```

File: ts/features/zendesk/screens/ZendeskAskPermissions.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    title: I18n.t("support.askPermissions.cta.allow")
+  };
+  return (
+    <BaseScreenComponent
+      showInstabugChat={false}
+      goBack={true}
+      customRightIcon={{
+        iconName: "",
+        onPress: () => true
+      }}
```
Comment (Undermaken) [inline ts/features/zendesk/screens/ZendeskAskPermissions.tsx:34]:
```suggestion
```
Maybe it couldn't be needed?

File: ts/features/zendesk/screens/ZendeskChooseCategory.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import I18n from "../../../i18n";
+
+const ZendeskChooseCategory = () => (
+    <BaseScreenComponent
+      showInstabugChat={false}
+      goBack={true}
+      customRightIcon={{
+        iconName: "",
+        onPress: () => true
+      }}
```
Comment (Undermaken) [inline ts/features/zendesk/screens/ZendeskChooseCategory.tsx:16]: Same of above

File: ts/navigation/AppNavigator.ts [hunk 4]
```diff
@@ -36,6 +38,9 @@ const navigator = createStackNavigator(
     },
     [ROUTES.WORKUNIT_GENERIC_FAILURE]: {
       screen: WorkunitGenericFailure
+    },
+    [ZENDESK_ROUTES.HELP_CENTER]: {
```
Comment (Undermaken) [inline ts/navigation/AppNavigator.ts:42]: we should add these screens only when the FF is enabled
see https://github.com/pagopa/io-app/blob/1c2d2b54b96ad5b075e702857f8517b1ea06759d/ts/navigation/WalletNavigator.ts#L222

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/navigation/AppNavigator.ts (hunk 4 above)
Comment (debiff) [reply ts/navigation/AppNavigator.ts:42]: Thank you, added in [51f6383](https://github.com/pagopa/io-app/pull/3538/commits/51f6383303aa5c25d656b9df10a89967a94cf997)

File: ts/components/ZendeskSupportComponent.tsx (hunk 1 above)
Comment (debiff) [reply ts/components/ZendeskSupportComponent.tsx]: Sorry, my fault! I copied-paste the previous code! Refactored in [ab99c4b](https://github.com/pagopa/io-app/pull/3538/commits/ab99c4bc9c7164d5579339a0e99a5d9108df4bb7)

File: ts/features/zendesk/navigation/navigator.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    }
+  : {};
+
+export const zendeskSupportNavigator = createStackNavigator(zendeskConfigMap, {
+  // Let each screen handles the header and navigation
+  headerMode: "none",
+  defaultNavigationOptions: {
+    gesturesEnabled: false
+  }
+});
```
Comment (Undermaken) [inline ts/features/zendesk/navigation/navigator.tsx:22]: It doesn't work in this way.
If you put `zendeskEnabled===false` the app doesn't start



I suggest to apply the same patter [used here](https://github.com/pagopa/io-app/blob/1c2d2b54b96ad5b075e702857f8517b1ea06759d/ts/navigation/WalletNavigator.ts#L222) for example
I tested this code by editing the file `ts/navigation/AppNavigator.ts`

```typescript
const configMap = {
  [ROUTES.INGRESS]: {
    // This is the first screen that gets loaded by the app navigator
    // On component mount, the screen will dispatch an
    // APPLICATION_INITIALIZED action that gets handled by the startup saga.
    screen: IngressScreen
  },
  [ROUTES.BACKGROUND]: {
    screen: BackgroundScreen
  },
  [ROUTES.AUTHENTICATION]: {
    // The navigator used during authentication
    screen: AuthenticationNavigator
  },
  [ROUTES.ONBOARDING]: {
    // The navigator user during onboarding for authenticated users
    screen: OnboardingNavigator
  },
  [ROUTES.MAIN]: {
    // The navigator used for authenticated users on onboarding completed
    screen: MainNavigator
  },
  [ROUTES.WORKUNIT_GENERIC_FAILURE]: {
    screen: WorkunitGenericFailure
  }
};

const zendeskMap = zendeskEnabled
  ? {
      [ZENDESK_ROUTES.HELP_CENTER]: {
        screen: zendeskSupportNavigator
      }
    }
  : {};

/**
 * The main stack of screens of the Application.
 */
const navigator = createStackNavigator(
  { ...configMap, ...zendeskMap },
  {
    // Let each screen handle the header and navigation
    headerMode: "none",
    defaultNavigationOptions: {
      gesturesEnabled: false
    }
  }
);

export default navigator;
```

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/features/zendesk/screens/ZendeskAskPermissions.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    testID: "continueButtonId",
+    bordered: false,
+    onPress: () => navigation.navigate(navigateToZendeskChooseCategory()),
+    title: I18n.t("support.askPermissions.cta.allow")
+  };
+  return (
+    <BaseScreenComponent
+      showInstabugChat={false}
+      goBack={true}
+      customRightIcon={{
```
[the remark this answers is not in the document]
Comment (debiff) [reply ts/features/zendesk/screens/ZendeskAskPermissions.tsx:32]: Sure, done in [22ad4fb](https://github.com/pagopa/io-app/pull/3538/commits/22ad4fb397e6e171bed5f1b5f0727b6c0c115eb7)

File: ts/features/zendesk/navigation/navigator.tsx (hunk 5 above)
Comment (debiff) [reply ts/features/zendesk/navigation/navigator.tsx]: Thank you! Critical! 
Fixed in [71ff382](https://github.com/pagopa/io-app/pull/3538/commits/71ff38266f4599e0cec703a570d6499eecfa5e5c)

**Review by Undermaken — APPROVED** [review]

**Pull request merged**